backend/windmill-worker/src/go_executor.rs

```diff
@@ -215,12 +210,11 @@ func Run(req Req) (interface{{}}, error){{
         )
         .await?;
 
-        if let Some(bin_path) = bin_path.clone() {
-            tokio::fs::copy(format!("{job_dir}/main"), &bin_path).await?;
-            logs.push_str(&format!("write cached binary: {}\n", bin_path));
-        }
+        create_dir(&bin_path).await?;
+        tokio::fs::copy(format!("{job_dir}/main"), format!("{bin_path}/main")).await?;
```

Comment:
And then, the `copy` expects 2 file paths. Doing `copy /path/to/main /dest/path/` won't work, you need to do `copy /path/to/main /dest/path/main`. Same below